### DISCUSSION TITLE
Not found errors still occur in finalizers

### DISCUSSION THREAD

**Discussion by moluzhui** [kubernetes-sigs/kubebuilder#2440] [Support] [answered]
I use finalizers to delete dependence, but it will occurs error of "unable to fetch CRD"

Although this will not be a problem because it is only connected to the current Reconciile, I still don't want any errors. How can I solve this problem

I really appreciate any help with this

**Comment by camilamacedo86** [comment, marked as answer]
Hi @moluzhui, 

Are you fetching the resource in the beginner of your reconcile as it is done in the example: https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v3/memcached-operator/controllers/memcached_controller.go#L62-L76?

Are you adding the finalizers inside of the logic to ensure that is only called when the resource is found? (e.g. https://sdk.operatorframework.io/docs/building-operators/golang/advanced-topics/#external-resources)

**Reply by moluzhui** [reply]
Thank you for you help, I followed kubebuilder's Cronjob example, which threw a NotFound error directly. I think I'll need to look at the operator-sdk example later

**Comment by camilamacedo86** [comment]
Closing since it is answered.
